src/services/codefixes/inferFromUsage.ts

```diff
@@ -262,11 +306,13 @@ namespace ts.codefix {
             }
             const isConstructor = declaration.kind === SyntaxKind.Constructor;
             const callContexts = isConstructor ? usageContext.constructContexts : usageContext.callContexts;
+            let isOptional = false;

```

Comment:
It passed all tests for me when I moved this inside the callback after `const isRest`.

Reply:
I was trying to be clever since once we've encountered one parameter past the end of a call, all the rest must be optional. But the information is actually recalculated &mdash; cheaply &mdash; for each parameter, so there's no point.